src/apps/patron-page/sections/PincodeSection.tsx

```diff
@@ -73,7 +73,11 @@ const PincodeSection: FC<PincodeSectionProps> = ({
           inputmode="numeric"
           onChange={(newPin) => setPincode(newPin)}
           value={pincode}
-          label={t("patronPagePincodeLabelText")}
+          label={
+            required
+              ? `${t("patronPagePincodeLabelText")}*`
```

Comment:
I think it should be the responsibility of the `TextInput` component to prefix label text with an asterisk if it is required.